src/encrypt/aescbc.rs

```diff
@@ -0,0 +1,200 @@
+            .arg("rand")
+            .arg("16")
+            .output()
+            .await
+            .context("failed to run openssl command")?;
+
+        if !output.status.success() {
+            return Err(anyhow::anyhow!("openssl command failed with status: {}", output.status));
+        }
+
+        Ok(output.stdout)
+    }
+}
+
+#[async_trait]
+impl Transformer for AesCbc {
+    fn get_prefix(&self) -> String {
+        self.prefix.to_string()
+    }
+
+    // openssl enc -d -aes-256-cbc \
+    //   -K <base64-decoded-key> \
+    //   -iv <initialization-vector> \
+    //   -in <ciphertext-file> \
+    //   -out <plaintext-file>
+    async fn decrypt(&self, _etcd_key: String, ciphertext: Vec<u8>) -> Result<Vec<u8>> {
+        let (_, cipher_data) = ciphertext.split_at(self.prefix.len());
+        ensure!(cipher_data.len() >= BLOCK_SIZE, "data is shorter than the required block size");
+
+        let (iv, data) = cipher_data.split_at(BLOCK_SIZE);
+        ensure!(data.len() % BLOCK_SIZE == 0, "invalid block size");
+
+        let key = base64_standard.decode(self.key.as_bytes())?;
+
+        let key_hex = hex::encode(key);
+        let iv_hex = hex::encode(iv);
+
+        let mut command = Command::new("openssl")
+            .arg("enc")
+            .arg("-d")
+            .arg("-aes-256-cbc")
+            .arg("-K")
+            .arg(&key_hex)
+            .arg("-iv")
+            .arg(&iv_hex)
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .context("failed to spawn openssl")?;
+
+        let stdin_future = async {
+            if let Some(mut stdin) = command.stdin.take() {
+                stdin.write_all(data).await.context("failed to write to stdin")?;
+            }
+
+            Ok::<_, anyhow::Error>(())
+        };
+
+        let mut plaintext = Vec::new();
+        let mut stderr = Vec::new();
+
+        let stdout_future = async {
+            if let Some(mut stdout) = command.stdout.take() {
+                stdout.read_to_end(&mut plaintext).await.context("failed to read from stdout")?;
+            }
+
+            Ok::<_, anyhow::Error>(())
+        };
+
+        let stderr_future = async {
+            if let Some(mut stderr_pipe) = command.stderr.take() {
+                stderr_pipe.read_to_end(&mut stderr).await.context("failed to read from stderr")?;
+            }
+
+            Ok::<_, anyhow::Error>(())
+        };
+
+        let (stdin_result, stdout_result, stderr_result) = tokio::join!(stdin_future, stdout_future, stderr_future);
+
+        stdin_result?;
+        stdout_result?;
+        stderr_result?;
+
+        let status = command.wait().await.context("failed to wait for openssl")?;
+
+        if status.success() {
+            Ok(plaintext)
+        } else {
+            bail!(format!("AES-CBC decryption error: {}", String::from_utf8_lossy(&stderr)));
+        }
```

Comment:
Awesome simplification, I was sure it wasn't as readable as it can be :)

But it just blocks, any idea what are we missing compared to the previous version?

Reply:
Hmm, don't know

Reply:
```rust
    let (mut command_stdin, mut command_stdout, mut command_stderr) = (
        command.stdin.take().context("failed to take stdin")?,
        command.stdout.take().context("failed to take stdout")?,
        command.stderr.take().context("failed to take stderr")?,
    );

    command_stdin.write_all(data).await.context("failed to write stdin")?;
    drop(command_stdin);

    command_stdout.read_to_end(&mut plaintext).await.context("failed to read stdout")?;
    command_stderr.read_to_end(&mut stderr).await.context("failed to read stderr")?;
```

This should be good enough

Reply:
This still blocks at some point, I think if we read concurrently `stdout` and `stderr` this issue will be solved.

Reply:
> This still blocks at some point, I think if we read concurrently stdout and stderr this issue will be solved.

No, it doesn't actually.